lib/datasource/index.ts

```diff
@@ -149,17 +148,16 @@ async function mergeRegistries(
 
 function resolveRegistryUrls(
   datasource: Datasource,
-  extractedUrls: string[]
+  extractedUrls: string[] = []
 ): string[] {
   const { defaultRegistryUrls = [], appendRegistryUrls = [] } = datasource;
-  const customUrls = extractedUrls?.filter(Boolean);
-  let registryUrls: string[];
-  if (is.nonEmptyArray(customUrls)) {
-    registryUrls = [...extractedUrls, ...appendRegistryUrls];
-  } else {
-    registryUrls = [...defaultRegistryUrls, ...appendRegistryUrls];
```

Comment:
The only reason for not included `defaultRegistryUrls` right before `extractedUrls` was because we did it in manager level?
Mabe I did't know something important when refactored this function...

Reply:
You need to be able to override defaults